File: pinpointB/src/main/java/com/nxquar/pinpoint/service/AddressService.java
package com.nxquar.pinpoint.service;

import com.nxquar.pinpoint.DTO.MessageResponse;
import com.nxquar.pinpoint.Model.Address;

import java.util.List;
import java.util.UUID;

public interface AddressService {
    Address createAddress (Address address,UUID instituteId);
    Address getAddresses(UUID instituteId);
    MessageResponse updateAddress(Address address);
    MessageResponse deleteAddress(UUID id);
}

File: pinpoint/lib/data/datasource/address/address_service.dart
import 'package:dio/dio.dart';
import 'package:pinpoint/model/address/address.dart';
import 'package:pinpoint/model/message_response.dart';
import 'package:pinpoint/resources/constant/string/appString.dart';
import 'package:retrofit/retrofit.dart';

part 'address_service.g.dart';

@RestApi(baseUrl: "${AppString.baseUrl}/api/addresses")
abstract class AddressService {
  factory AddressService(Dio dio, {String baseUrl}) = _AddressService;

  @POST("/{instituteId}")
  Future<HttpResponse<Address>> createAddress(
    @Path("instituteId") String instituteId,
    @Body() Address address,
  );

  @GET("/{instituteId}")
  Future<HttpResponse<Address>> getAddress(@Path("instituteId") String instituteId);

  @PUT("")
  Future<HttpResponse<MessageResponse>> updateAddress(@Body() Address address);

  @DELETE("/{id}")
  Future<HttpResponse<MessageResponse>> deleteAddress(@Path("id") String id);
}
